File: scala-demo/src/main/scala/demo1/Ex1.scala
package demo1

import akka.actor.ActorSystem
import akka.stream.ActorMaterializer
import akka.stream._
import akka.stream.scaladsl._

import scala.concurrent.Future
import scala.concurrent.Await
import scala.concurrent.duration.Duration
import scala.concurrent.duration._

object Ex1 extends App {
  implicit val system = ActorSystem("demo1")
  implicit val materializer = ActorMaterializer()
  import system.dispatcher

  // Create a source with some integer and print them
  val source = Source(List(1,2,3))
  val sink = Sink.foreach(println)
  val runnable: RunnableGraph[Unit] = source.to(sink)
  runnable.run()
}

File: java-demo/src/main/java/demo1/Ex1.java
package demo1;

import java.util.Arrays;

import scala.concurrent.Future;
import scala.runtime.BoxedUnit;
import akka.actor.ActorSystem;
import akka.stream.ActorMaterializer;
import akka.stream.javadsl.RunnableGraph;
import akka.stream.javadsl.Sink;
import akka.stream.javadsl.Source;

public class Ex1 {
  public static void main(String[] args) {
    ActorSystem system = ActorSystem.create("demo1");
    ActorMaterializer materializer = ActorMaterializer.create(system);

    // Create a source with some integer and print them
    Source<Integer, BoxedUnit> source = Source.from(Arrays.asList(0,1,2,3));
    Sink<Integer, Future<BoxedUnit>> sink = Sink.foreach(x -> System.out.println(x));
    RunnableGraph<BoxedUnit> runnable = source.to(sink);
    runnable.run(materializer);
  }
}
